packages/amplify-e2e-core/src/categories/api.ts

```diff
@@ -298,3 +298,244 @@ export function addRestApi(cwd: string, settings: any) {
     }
   });
 }
+
+//add default api
+export function addApiWithAPIKeyAuthType(projectDir: string) {
+  return new Promise((resolve, reject) => {
+    spawn(getCLIPath(), ['add', 'api'], { cwd: projectDir, stripColors: true })
+      .wait('Please select from one of the below mentioned services:')
+      .sendCarriageReturn()
+      .wait('Provide API name:')
+      .sendCarriageReturn()
+      .wait('Choose the default authorization type for the API')
+      .sendCarriageReturn()
+      .wait('Enter a description for the API key')
+      .sendCarriageReturn()
+      .wait('After how many days from now the API key should expire')
+      .sendCarriageReturn()
+      .wait('Do you want to configure advanced settings for the GraphQL API')
+      .sendCarriageReturn()
+      .wait('Do you have an annotated GraphQL schema?')
+      .sendLine('n')
+      .wait('Do you want a guided schema creation')
+      .sendLine('y')
+      .wait('What best describes your project')
+      .sendCarriageReturn()
+      .wait('Do you want to edit the schema now?')
+      .sendLine('n')
+      .wait('"amplify publish" will build all your local backend and frontend resources')
+      .run((err: Error) => {
+        if (!err) {
+          resolve();
+        } else {
+          reject(err);
+        }
+      });
+  });
+}
```

Comment:
How is this different from the function below except for not passing an Schema file? Wouldn't it be possible to update this function to accept optional schema file and have both functionality?  https://github.com/aws-amplify/amplify-cli/blob/c522f295304410aeb1d6f60aaba9b466d3304ee1/packages/amplify-e2e-core/src/categories/api.ts#L47-L78

There is also another function which does the exact same thing
https://github.com/aws-amplify/amplify-cli/blob/c522f295304410aeb1d6f60aaba9b466d3304ee1/packages/amplify-e2e-core/src/categories/api.ts#L11-L46

Reply:
That particular function is removed from this PR. The add Api workflows with different auth settings in the PR are combined into a single function.